discovery/source/DiscoveryClient.cpp

```diff
@@ -91,32 +91,33 @@ namespace Aws
                     std::shared_ptr<Crt::Http::HttpClientConnection> connection, int errorCode) {
                     if (!errorCode)
                     {
+                        auto request = Aws::Crt::MakeShared<Crt::Http::HttpRequest>(m_allocator, m_allocator);
```

Comment:
there are a few places here that can fail with OOM. This MakeShared(), also the SetMethod(), SetPath(), and AddHeader() below